packages/components/addon/components/hds/dropdown/list-item/copy-item.js

```diff
@@ -37,12 +38,21 @@ export default class HdsDropdownListItemCopyItemComponent extends Component {
   }
 
   @action
-  copyCode() {
-    navigator.clipboard.writeText(this.args.text);
-    // this if statement resolves to [object Promise] so maybe some improvements
-    // could be made here
-    if (navigator.clipboard.readText()) {
-      this.isSuccess = true;
+  async copyCode() {
+    // If there is text to be copied, then copy it to the clipboard.
+    // If the text in the clipboard is the same as the text to be copied, then set isSuccess to true.
+    // TODO: set any _other_ isSuccess to false, so there is only one indicated successful copy at a time.
```

Comment:
[suggestion] what if instead we just set a timeout that resets the status?

Reply:
Yes, perhaps! Looking into it as part of a different issue though: https://github.com/hashicorp/design-system/issues/268